source/server/connection_handler_impl.cc

```diff
@@ -13,7 +14,9 @@ namespace Envoy {
 namespace Server {
 
 ConnectionHandlerImpl::ConnectionHandlerImpl(spdlog::logger& logger, Api::ApiPtr&& api)
-    : logger_(logger), api_(std::move(api)), dispatcher_(api_->allocateDispatcher()) {}
+    : logger_(logger), api_(std::move(api)), dispatcher_(api_->allocateDispatcher()) {
+  UNREFERENCED_PARAMETER(logger_);
```

Comment:
Yes, the compiler complains in the GLOG_ON case.  Is there a preferred way of ignoring the member?  We could make another macro for unused members that puts a compiler directive in front of it where it's declared.